src/Microsoft.OpenApi/Models/OpenApiRequestBody.cs

```diff
@@ -102,15 +102,16 @@ public IOpenApiParameter ConvertToBodyParameter(IOpenApiWriter writer)
                 // V2 spec actually allows the body to have custom name.
                 // To allow round-tripping we use an extension to hold the name
                 Name = "body",
-                Schema = Content.Values.FirstOrDefault()?.Schema ?? new OpenApiSchema(),
-                Examples = Content.Values.FirstOrDefault()?.Examples,
+                Schema = Content?.Values.FirstOrDefault()?.Schema ?? new OpenApiSchema(),
+                Examples = Content?.Values.FirstOrDefault()?.Examples,
                 Required = Required,
-                Extensions = Extensions.ToDictionary(static k => k.Key, static v => v.Value)  // Clone extensions so we can remove the x-bodyName extensions from the output V2 model.
+                Extensions = Extensions?.ToDictionary(static k => k.Key, static v => v.Value)
             };
-            if (bodyParameter.Extensions.ContainsKey(OpenApiConstants.BodyName))
+            // Clone extensions so we can remove the x-bodyName extensions from the output V2 model.
+            if (bodyParameter.Extensions != null && bodyParameter.Extensions.TryGetValue(OpenApiConstants.BodyName, out var bodyNameExtension))
```

Comment:
test the type in the condition with is (type assertion) instead.
reduces allocations, and now you know bodyname can't be null this way